### PR TITLE
bsp: stm32: rename stm32mp1 config files

### PR DESCRIPTION
**Pull request by igoropaniuk**
Rename stm32mp1->stm32mp15, so it corresponds to the new naming in
meta-st-stm32mp layer.

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>

### CODE REVIEW

**Review by MrCry0 — APPROVED** [review]

**Review by ricardosalveti — APPROVED** [review]
LGTM, once merged just backport to 5.15 as we don't need to keep 5.10 in sync since it won't be used by this target anymore.

**Pull request merged**
